doc/lexical_envs.rst

```diff
@@ -0,0 +1,265 @@
+********************
+Lexical environments
+********************
+
+Lexical environments make it possible to create maps from symbols to nodes:
+they are the most crucial data structure used to implement semantic analysis
+with Langkit.  There are various ways to link lexical environments together,
+described below, so that a symbol lookup through a single lexical environment
+can actually perform a search through others envs.
+
+Nodes have either one or two associated lexical environments, available through
```

Comment:
This should probably be *after* the example, using an example with the node structure you proposed here. Here I think it's a bit unclear, since you didn't really see yet what it looks like.

Reply:
I’ve inserted mininal examples here to clarify this: I think the last example is too big and uses too many notions to appear before explanations. I’ve added a minimal one at the beginning, and moved this part right before the big example. Please let me know if this is OK for you.